tpu/hptuning/resnet-hptuning/resnet/resnet_main.py

```diff
@@ -0,0 +1,512 @@
+        # Since the coordinator is on a different job than the TPU worker,
+        # sometimes the TPU worker does not finish initializing until long after
+        # the CPU job tells it to start evaluating. In this case, the checkpoint
+        # file could have been deleted already.
+        tf.logging.info(
+            'Checkpoint %s no longer exists, skipping checkpoint' % ckpt)
+
+  else:   # FLAGS.mode == 'train' or FLAGS.mode == 'train_and_eval'
+    current_step = estimator._load_global_step_from_checkpoint_dir(FLAGS.model_dir)  # pylint: disable=protected-access,line-too-long
+    batches_per_epoch = NUM_TRAIN_IMAGES / FLAGS.train_batch_size
+    tf.logging.info('Training for %d steps (%.2f epochs in total). Current'
+                    ' step %d.' % (FLAGS.train_steps,
+                                   FLAGS.train_steps / batches_per_epoch,
+                                   current_step))
+
+    start_timestamp = time.time()  # This time will include compilation time
+    if FLAGS.mode == 'train':
+      tf.logging.info('Training for trial_{}'.format(trial_id))
+      resnet_classifier.train(
+          input_fn=imagenet_train.input_fn, max_steps=FLAGS.train_steps)
+
+    else:
+      assert FLAGS.mode == 'train_and_eval'
+      while current_step < FLAGS.train_steps:
+        # Train for up to steps_per_eval number of steps.
+        # At the end of training, a checkpoint will be written to --model_dir.
+        next_checkpoint = min(current_step + FLAGS.steps_per_eval,
+                              FLAGS.train_steps)
+        tf.logging.info('Training for trial_{}'.format(trial_id))
+        resnet_classifier.train(
+            input_fn=imagenet_train.input_fn, max_steps=next_checkpoint)
+        current_step = next_checkpoint
+
+        # Evaluate the model on the most recent model in --model_dir.
+        # Since evaluation happens in batches of --eval_batch_size, some images
+        # may be consistently excluded modulo the batch size.
+        tf.logging.info('Starting to evaluate.')
+        tf.logging.info('Evaluating for trial_{}'.format(trial_id))
+        eval_results = resnet_classifier.evaluate(
+            input_fn=imagenet_eval.input_fn,
+            steps=NUM_EVAL_IMAGES // FLAGS.eval_batch_size)
+        tf.logging.info('Eval results: %s' % eval_results)
+
+    elapsed_time = int(time.time() - start_timestamp)
+    tf.logging.info('Finished training up to step %d. Elapsed seconds %d.' %
+                    (FLAGS.train_steps, elapsed_time))
+
+    if FLAGS.export_dir is not None:
+      # The guide to serve a exported TensorFlow model is at:
+      #    https://www.tensorflow.org/serving/serving_basic
+      tf.logging.info('Starting to export model.')
+      resnet_classifier.export_savedmodel(
+          export_dir_base=FLAGS.export_dir,
+          serving_input_receiver_fn=imagenet_input.image_serving_input_fn)
+
+if __name__ == '__main__':
+  tf.logging.set_verbosity(tf.logging.INFO)
+  tf.app.run()
```

Comment:
change tf.app.run() to simply python call as main(...). No need for this invocation.